src/robot/libraries/Process.py

```diff
@@ -439,11 +442,15 @@ def wait_for_process(self, handle=None, timeout=None, on_timeout='continue'):
         | ${result} =                 | Wait For Process | timeout=1min 30s | on_timeout=kill |
         | Process Should Be Stopped   |                  |                  |
         | Should Be Equal As Integers | ${result.rc}     | -9               |
+        | # Never-ending process      |                  |                  |
+        | ${result} =                 | Wait For Process | timeout=0        | on_timeout=kill |
+        | Process Should Be Stopped   | # Will return ``Fail``              |
         """
         process = self._processes[handle]
         logger.info('Waiting for process to complete.')
-        if timeout:
-            timeout = timestr_to_secs(timeout)
+        secs = timestr_to_secs(timeout)
+        if is_truthy(timeout) and secs > 0:
+            timeout = secs
```

Comment:
I'd prefer this:
```python
timeout = timestr_to_secs(timeout) if timeout else 0
if timeout > 0:
    # ...
```

Reply:
Unfortunately, this won't handle

> `NONE` (case-insensitively) is same as not specifying timeout at all.

I'll merge your proposal with my changes to support the aforementioned statement.